design/batch_retrieval_design.md

```diff
@@ -0,0 +1,20 @@
+# Batch retrieval design
+
+This document will cover the options for enhancing Batch Retrieval requests.
+
+### Motivation
+
+*Previously*, for the init container solution, each Secrets Provider was sitting within the same Deployment as the app. Because of this, when a batch retrieval request was made, the requests were dispersed across many call to the Conjur server. If one K8s Secret was not able to be updated with a Conjur value (for example due to a permission error), then the batch request would fail and only that pod would not spin up successfully. 
+
+*Now* that the Secrets Provider is outside, a single batch retrieval request will be made to Conjur for *all* application. So if there is a failure in receiving one of the values, then the whole request would fail. In other words, **no application** will spin up in the namespace.
```

Comment:
I don't see any reference to the fact that the authn-k8s which secrets provider use will not work when using it from out side the cluster as @oburstein-hub wrote [here](https://ca-il-confluence.il.cyber-ark.com/display/rndp/K8S+Secret+provider+for+HomeDepot+Solution+Architecture+Issues#/) when using option number 3 we need access to the k8s api from the follower.
Please address the limitations from the doc in this design

Reply:
@Tovli - lets wait with this to this subject closure with Inbal / Naama.

Reply:
As talked with @InbalZilberman - We need to support the secrets provider being outside the follower cluster.
This is actually not new, since we needed to support it also before we had the new requirements

Reply:
@Tovli I don't understand.
This requirement was never introduced as part of this solution, and is unrelated to anything we wrote here.
@InbalZilberman Added this to the [feature doc](https://github.com/cyberark/secrets-provider-for-k8s/issues/102) only yesterday as a minor sentence for testing.
This is not minor, and it is way more than testing, since testing is not a customer requirement; supporting is.
This need to be handled according to regular R&D requirements, and not as part of this solution design.
Just a reminder, the requirements here are support multiple apps, deploy outside of app lifecycle, and support rotation with seamless upgrade between milestones.
These are all Secrets Provider requirements.
There's nothing regarding follower, much less it's location and availability.

Reply:
I strongly agree with @abrahamko on this